File: 0009-palindrome-number/0009-palindrome-number.py
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        rev = 0
        a=x
        if(x>0):
            while(x>0):
                rem = x%10
                rev = rev*10+rem
                x//=10
            if(rev==a):
                return True
        elif(x==0):
            return True
        else:
            return False
        

File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        int num=0;
        int original = x;
        if(x>0){
            while(x>0){
                int r = x%10;
                if(num>(INT_MAX-r)/10){
                    return 0;
                }
                num=num*10+r;
                x=x/10;
            }
        }
        if(num==original){
            return true;
        }
        else{
            return false;
        }
}
};